Source/WebCore/platform/graphics/gstreamer/ImageDecoderGStreamer.cpp

```diff
@@ -39,13 +38,25 @@ GST_DEBUG_CATEGORY(webkit_image_decoder_debug);
 #define GST_CAT_DEFAULT webkit_image_decoder_debug
 
 static Lock s_decoderLock;
-static Vector<RefPtr<ImageDecoderGStreamer>> s_imageDecoders;
+static Vector<WeakPtr<ImageDecoderGStreamer>> s_imageDecoders;
```

Comment:
I don't think this is thread safe, the vector has a lock, but WeakPtr need to be destroyed in the same thread it was created. You probably need to use ThreadSafeWeakPtr, and maybe instead of a Vector, use a ThreadSafeWeakHashSet if the order doesn't matter.

Reply:
I'll revert these changes. ThreadSafeWeakPtr can't be used because the ImageDecoder parent class is already ThreadSafeRefCounted, I don't see how this would work with a ThreadSafeWeakHashSet storage.